### PR TITLE
feat: support datetime format `%Y-%m-%dT%H:%M:%S%.f%z`

### LINKED ISSUE #6947: tasks:support cast '2022-02-01T00:00:00+08:00'  into timestamp
tasks:
- [ ] support cast '2022-02-01T00:00:00+08:00'  into timestamp

_Originally posted by @sundy-li in https://github.com/datafuselabs/databend/issues/6522#issuecomment-1176983810_

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

support a new timestamp format.

Fixes #6947

### CODE REVIEW

File: tests/logictest/suites/gen/02_function/02_0012_function_datetimes [hunk 1]
```diff
@@ -103,87 +103,72 @@ select to_datetime('0999-12-31 23:59:59');
 ----
 1000-01-01 00:00:00.000000
 
+statement query T
+select to_datetime('2022-12-31T23:59:59+00:00');
+
+----
+2022-12-31 23:59:59.000000
```
Comment (b41sh) [inline tests/logictest/suites/gen/02_function/02_0012_function_datetimes:111]: how about adding some test cases for other timezones?

Comment (TCeason) [reply tests/logictest/suites/gen/02_function/02_0012_function_datetimes:111]: Yes, in this test file:tests/logictest/suites/gen/11_data_type/11_0001_data_type_date_time

File: common/functions/src/scalars/expressions/cast_from_string.rs [hunk 2]
```diff
@@ -121,7 +126,22 @@ pub fn string_to_timestamp(date_str: impl AsRef<[u8]>, tz: &Tz) -> Option<DateTi
                         Some(dt)
                     }
                 }
-                Err(_) => None,
+                Err(_) => {
+                    // DateTime::parse_from_str requires that it successfully parse an entire DateTime -- full date and time and timezone.
```
Comment (sundy-li) [inline common/functions/src/scalars/expressions/cast_from_string.rs:130]: `YYYY-MM-DD` should be considered.

**Comment by BohuTANG** [comment]
Conflicting files :/

**Comment by TCeason** [comment]
> Conflicting files :/

Got. And because this pr is not elegant enough, consider reimplementing it.

**Comment by Xuanwo** [comment]
> because this pr is not elegant enough, consider reimplementing it.

Let's go!

**Comment by Xuanwo** [comment]
Let's close this PR now. Feel free to start a new PR when you are ready!

**Pull request closed without merging**